doc/user_guide/recordflux_ug/03-getting_started_with_recordflux.rst

```diff
@@ -0,0 +1,743 @@
+
+      -- Message is not yet valid
+      if not Valid_Message (Ctx) then
+         Ada.Text_IO.Put_Line ("Expected");
+      else
+         Ada.Text_IO.Put_Line ("Unexpected");
+      end if;
+
+      Set_Tag    (Ctx, RFLX.TLV.Msg_Data);
+      Set_Length (Ctx, 4);
+      Set_Value  (Ctx, (10, 20, 30, 40));
+
+      -- Verify field status
+      if Valid (Ctx, F_Tag)
+         and Valid (Ctx, F_Length)
+         and Well_Formed (Ctx, F_Value)
+         and not Valid (Ctx, F_Value)
+      then
+         Ada.Text_IO.Put_Line ("Expected");
+      else
+         Ada.Text_IO.Put_Line ("Unexpected");
+      end if;
+
+      --  Inspect message contents
+      Take_Buffer (Ctx, Buffer);
+      if Buffer.all = (1, 0, 4, 10, 20, 30, 40) then
+         Ada.Text_IO.Put_Line ("Expected");
+      else
+         Ada.Text_IO.Put_Line ("Unexpected");
+      end if;
+   end Generate_TLV;
+
+Note that, as an opaque field, the ``Value`` field is
+well-formed but not valid (lines 30 and 31),
+since it has not been refined.
+
+Executing the program produces the following output:
+   
+.. code:: console
+
+   $ ./generate_tlv
+   Expected
+   Expected
+   Expected
+
+Parsing a Message
+----------------------
+
+The following program decomposes a TLV message into its
+constituent fields:
+
+.. doc-check: ignore
+.. code:: ada
+   :number-lines:
+
+   with Ada.Text_IO; use Ada.Text_IO;
+   with RFLX.RFLX_Types;
+   with RFLX.RFLX_Builtin_Types;
+   with RFLX.TLV.Message; use RFLX.TLV.Message;
+   procedure Parse_TLV is
+
+      Buffer  : RFLX.RFLX_Builtin_Types.Bytes_Ptr := 
+        new RFLX.RFLX_Builtin_Types.Bytes'(1, 0, 4, 10, 20, 30, 40);
+      Ctx       : Context;
+      Tag       : RFLX.TLV.Tag_Type;
+      Length    : RFLX.TLV.Length_Type;
+      Value_Ref : RFLX.RFLX_Builtin_Types.Bytes_Ptr;
+
+   begin
+      Initialize (Ctx, 
+                  Buffer,
+                  Written_Last => 
+                     RFLX.RFLX_Types.To_Last_Bit_Index (Buffer'Last));
+ 
+      Verify_Message (Ctx);
+
+      if Well_Formed_Message (Ctx) then
+         Tag := Get_Tag (Ctx);
+         Put_Line ("Tag field is " & Tag'Img);
+         case Tag is
+            when RFLX.TLV.Msg_Data =>
+               Length := Get_Length (Ctx);
+               Put_Line( "Length field is " & Length'Img);
+               if Present (Ctx, F_Value) then
+                  Value_Ref := 
+                     new RFLX.RFLX_Types.Bytes'( Get_Value(Ctx) );
+                  Put ("Value field is ( ");
+                  for Byte_Val of Value_Ref.all loop
+                     Put( Byte_Val'Img & ' ');
+                  end loop;
+                  Put_Line ( ")" ); 
+               else
+                  Ada.Text_IO.Put_Line ("Data message without value");
+               end if;
+            when RFLX.TLV.Msg_Error =>
+               Ada.Text_IO.Put_Line ("Error message");
+         end case;
+      else
+         Ada.Text_IO.Put_Line ("Invalid message");
+      end if;
+   end Parse_TLV;
```

Comment:
While looking at #1241, we realized that code blocks containing main functions with names different from `Main` are already supported. The shown warning can be safely ignored. The code block can be successfully checked, if some errors are fixed:

- The types of TLV do not have a `_Type` suffix. (The files in `tests/spark/generated` are used for compilation. Most of the files are generated from the specifications in `tests/data/specs/`.)
- `Well_Formed_Message` is still called `Structural_Valid_Message`.
- The function `Get_Value` is a ghost entity and thus cannot be used in this way. (This function is just intended for SPARK contracts. The `Get_Value` function would require a secondary stack, but we aim to support platforms where no secondary stack is allowed.) There is a procedure with the same name which can be used instead.

I would also recommend to always use `RFLX.RFLX_Types` instead `RFLX.RFLX_Builtin_Types` in all examples. There is no need to use both.

Here is the complete diff:

```diff
-.. doc-check: ignore
 .. code:: ada
    :number-lines:
 
    with Ada.Text_IO; use Ada.Text_IO;
    with RFLX.RFLX_Types;
-   with RFLX.RFLX_Builtin_Types;
    with RFLX.TLV.Message; use RFLX.TLV.Message;
+
    procedure Parse_TLV is
 
-      Buffer  : RFLX.RFLX_Builtin_Types.Bytes_Ptr := 
-        new RFLX.RFLX_Builtin_Types.Bytes'(1, 0, 4, 10, 20, 30, 40);
+      use type RFLX.RFLX_Types.Index;
+
+      Buffer  : RFLX.RFLX_Types.Bytes_Ptr :=
+        new RFLX.RFLX_Types.Bytes'(1, 0, 4, 10, 20, 30, 40);
       Ctx       : Context;
-      Tag       : RFLX.TLV.Tag_Type;
-      Length    : RFLX.TLV.Length_Type;
-      Value_Ref : RFLX.RFLX_Builtin_Types.Bytes_Ptr;
+      Tag       : RFLX.TLV.Tag;
+      Length    : RFLX.TLV.Length;
+      Value_Ref : RFLX.RFLX_Types.Bytes_Ptr := new RFLX.RFLX_Types.Bytes'(RFLX.RFLX_Types.Index'First .. RFLX.RFLX_Types.Index'First + 1024 => 0);
 
    begin
       Initialize (Ctx, 
@@ -685,7 +686,7 @@ constituent fields:
  
       Verify_Message (Ctx);
 
-      if Well_Formed_Message (Ctx) then
+      if Structural_Valid_Message (Ctx) then
          Tag := Get_Tag (Ctx);
          Put_Line ("Tag field is " & Tag'Img);
          case Tag is
@@ -693,8 +694,7 @@ constituent fields:
                Length := Get_Length (Ctx);
                Put_Line( "Length field is " & Length'Img);
                if Present (Ctx, F_Value) then
-                  Value_Ref := 
-                     new RFLX.RFLX_Types.Bytes'( Get_Value(Ctx) );
+                  Get_Value (Ctx, Value_Ref.all);
                   Put ("Value field is ( ");
                   for Byte_Val of Value_Ref.all loop
                      Put( Byte_Val'Img & ' ');
```